src/openrct2/object/SceneryBoundingBox.hpp

```diff
@@ -0,0 +1,304 @@
+            CoordsXY relativeCentroid = box.offset * 2 + box.length - rotationCenter * 2;
+            CoordsXY rotatedCentroid = { relativeCentroid.y, -relativeCentroid.x };
+            CoordsXY newCorner = (rotatedCentroid - rotatedLength) / 2 + rotationCenter;
+            return { { newCorner.x, newCorner.y, box.offset.z }, rotatedLength };
+        }
+
+        static SceneryBoundBoxes ReadBoundBoxes(json_t& jBBox, int32_t defaultHeight, bool fullTile)
+        {
+            SceneryBoundBoxes boxes;
+            if (jBBox.is_array())
+            {
+                Guard::Assert(jBBox.size() >= 4, "boundBox arrays require four elements, one for each view angle");
+                // array of four bboxes
+                for (uint8_t i = 0; i < NumOrthogonalDirections; i++)
+                    boxes[i] = ReadBoundBox(jBBox[i]);
+            }
+            else if (jBBox.is_object())
+            {
+                // single box, rotated around (16, 16) if fulltile or (8,8) if quarter tile
+                CoordsXY rotationCenter = { 8, 8 };
+                if (fullTile)
+                {
+                    rotationCenter = { 16, 16 };
+                }
+                auto bBox = ReadBoundBox(jBBox);
+                boxes[0] = bBox;
+                boxes[1] = RotateBoundBox(bBox, rotationCenter);
+                boxes[2] = RotateBoundBox(boxes[1], rotationCenter);
+                boxes[3] = RotateBoundBox(boxes[2], rotationCenter);
+            }
+            else
+            {
+                Guard::Assert(
+                    jBBox.is_string(),
+                    "boundBox must be an array of four boundBox objects, a single boundBox object, or a string matching the "
+                    "DefaultBoundingBoxType enum.");
+                boxes = GetDefaultSceneryBoundBoxes(GetBoundingBoxTypeFromString(Json::GetString(jBBox)));
+                for (uint8_t i = 0; i < NumOrthogonalDirections; i++)
+                    boxes[i].length.z = defaultHeight;
+            }
+            return boxes;
+        }
+
+        static const EnumMap<DefaultSpriteOffsetType> SpriteOffsetLookup{
+            { "quarterTile", DefaultSpriteOffsetType::QuarterTileOffset },
+            { "fullTileThin", DefaultSpriteOffsetType::FullTileThinOffset },
+            { "halfTile", DefaultSpriteOffsetType::FullTileOffset },
+            { "fullTile", DefaultSpriteOffsetType::FullTileOffset },
+            { "fullTileLarge", DefaultSpriteOffsetType::FullTileLargeOffset },
+            { "largeScenery", DefaultSpriteOffsetType::LargeSceneryOffset },
+        };
+
+        static DefaultSpriteOffsetType GetSpriteOffsetTypeFromString(const std::string& s)
+        {
+            auto result = SpriteOffsetLookup.find(s);
+            return (result != SpriteOffsetLookup.end()) ? result->second : DefaultSpriteOffsetType::LargeSceneryOffset;
+        }
+
+        static CoordsXYZ GetDefaultSpriteOffset(DefaultSpriteOffsetType type)
```

Comment:
I don't know why clang-tidy is flagging these as unused as they are used in SmallSceneryObject.cpp and LargeSceneryObject.cpp.

Reply:
Probably because you’re declaring them as static.